workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/HyscaleDeployServiceCommand.java

```diff
@@ -138,26 +143,32 @@ public void run() {
             // clean up service dir before dockerfileGen
             workflowContext.addAttribute(WorkflowConstants.CLEAN_UP_SERVICE_DIR, true);
 
-            // Generate Docker file
-            dockerfileGeneratorComponentInvoker.execute(workflowContext);
-
-            if (!workflowContext.isFailed()) {
-                imageBuildComponentInvoker.execute(workflowContext);
-            }
-
-            if (!workflowContext.isFailed()) {
-                manifestGeneratorComponentInvoker.execute(workflowContext);
-            }
-
-            if (!workflowContext.isFailed()) {
+            isServiceFailed = isServiceFailed ? isServiceFailed : !executeInvoker(dockerfileGeneratorComponentInvoker, workflowContext);
```

Comment:
Add a TODO, we need to overcome this kind of code. We need to register a callback to this and handle all these delta processing of invoker these itself.